forgerock-auth/src/main/java/org/forgerock/android/auth/callback/DeviceSigningVerifierCallback.kt

```diff
@@ -115,27 +110,37 @@ open class DeviceSigningVerifierCallback : AbstractCallback, Binding {
     }
 
     /**
-     * Sign the device.
+     * Sign the challenge with bounded device keys.
      *
      * @param context  The Application Context
-     * @param listener The Listener to listen for the result
+     * @param deviceAuthenticator A function to return an authenticator
      */
-    open suspend fun sign(context: Context) {
-        execute(context)
+    open suspend fun sign(context: Context,
+                          userKeySelector: UserKeySelector = DefaultUserKeySelector(),
+                          deviceAuthenticator: (type: DeviceBindingAuthenticationType) -> DeviceAuthenticator = {
+                              getDeviceAuthenticator(it)
+                          }) {
+        execute(context, userKeySelector = userKeySelector, deviceAuthenticator = deviceAuthenticator)
     }
 
 
     /**
-     * Bind the device.
+     * Sign the challenge with bounded device keys.
      *
      * @param context  The Application Context
+     * @param deviceAuthenticator A function to return an authenticator
      * @param listener The Listener to listen for the result
      */
-    open fun sign(context: Context, listener: FRListener<Void>) {
+    @JvmOverloads
+    open fun sign(context: Context,
+                  userKeySelector: UserKeySelector = DefaultUserKeySelector(),
+                  deviceAuthenticator: (type: DeviceBindingAuthenticationType) -> DeviceAuthenticator = {
```

Comment:
i always like to move the lambda implementation out of the function parameter ...it my choice..i see some people like to keep as it is.. so i am ok with either approach

Example 

 private val deviceAuthenticatorClosure: (type: DeviceBindingAuthenticationType) -> DeviceAuthenticator = {
        getDeviceAuthenticator(it)
    }

open fun sign(context: Context,
                  userKeySelector: UserKeySelector = DefaultUserKeySelector(),
                  deviceAuthenticator: (type: DeviceBindingAuthenticationType) -> DeviceAuthenticator = deviceAuthenticatorClosure)


val

Reply:
Since the implementation of each function is identical, I agree with u that we should extract it into a variable.